pkg/codegen/jenny_docs.go

```diff
@@ -322,42 +323,91 @@ func resolveInSchemaReference(ref string, root *simplejson.Json) (*schema, error
 	return &sch, nil
 }
 
+type mdSection struct {
+	title       string
+	extends     string
+	description string
+	rows        [][]string
+}
+
+func (md mdSection) write(w io.Writer) {
+	if md.title != "" {
+		fmt.Fprintf(w, "### %s\n", strings.Title(md.title))
+		fmt.Fprintln(w)
+	}
+
+	if md.description != "" {
+		fmt.Fprintln(w, md.description)
+		fmt.Fprintln(w)
+	}
+
+	if md.extends != "" {
+		fmt.Fprintln(w, md.extends)
+		fmt.Fprintln(w)
+	}
+
+	table := tablewriter.NewWriter(w)
+	table.SetHeader([]string{"Property", "Type", "Required", "Description"})
+	table.SetBorders(tablewriter.Border{Left: true, Top: false, Right: true, Bottom: false})
+	table.SetCenterSeparator("|")
+	table.SetAutoFormatHeaders(false)
+	table.SetHeaderAlignment(tablewriter.ALIGN_LEFT)
+	table.SetAutoWrapText(false)
+	table.AppendBulk(md.rows)
+	table.Render()
+	fmt.Fprintln(w)
+}
+
 // Markdown returns the Markdown representation of the schema.
 //
 // The level argument can be used to offset the heading levels. This can be
 // useful if you want to add the schema under a subheading.
-func (s schema) Markdown(level int) string {
-	if level < 1 {
-		level = 1
-	}
-
+func (s *schema) Markdown() string {
 	buf := new(bytes.Buffer)
 
-	if s.Title != "" && s.AdditionalProperties == nil {
-		fmt.Fprintf(buf, "### %s\n", strings.Title(s.Title))
-		fmt.Fprintln(buf)
+	var sections []mdSection
+	s.toSections(&sections)
+
+	for _, v := range sections {
+		v.write(buf)
 	}
 
-	if s.Description != "" {
-		fmt.Fprintln(buf, s.Description)
-		fmt.Fprintln(buf)
+	return buf.String()
+}
+
+func (s *schema) toSections(sections *[]mdSection) {
+	md := mdSection{}
+
+	if s.AdditionalProperties == nil {
+		md.title = s.Title
 	}
+	md.description = s.Description
 
 	if len(s.extends) > 0 {
-		fmt.Fprintln(buf, makeExtends(s.extends))
-		fmt.Fprintln(buf)
+		md.extends = makeExtends(s.extends)
 	}
+	md.rows = makeRows(s)
 
-	printProperties(buf, &s)
+	if !contains(sections, md) {
+		*sections = append(*sections, md)
```

Comment:
I feel it'd be a bit more natural (to read, imho) if the function would return the built sections instead of receiving the list as an argument, and modify it.